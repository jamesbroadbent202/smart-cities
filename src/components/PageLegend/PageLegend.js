import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../Icon/Icon';
import getColorVariant from '../../helpers/getColorVariant';
import style from './PageLegend.scss';

const PageLegend = (props) => {
  if (props.isAllCitiesPage && !props.cities) {
    console.warn('You said this is an all-cities page, but did not pass in any cities');

    return null;
  }

  const lightColor = getColorVariant(props.colorName, '020');
  const mediumColor = getColorVariant(props.colorName, '080');
  const darkColor = getColorVariant(props.colorName, '900');

  return (
    <div
      className={style.wrapper}
      style={{ background: lightColor }}
    >
      <div className={style.container}>
        {!props.isContextPage && (
          <div className={style.indicatorTypesWrapper}>
            <div className={style.indicatorTypeWrapper}>
              <Icon
                className={style.icon}
                color={darkColor}
                icon="contextualIndicator"
                size={24}
              />

              <span className={style.text}>Contextual indicators</span>
            </div>

            <div className={style.indicatorTypeWrapper}>
              <Icon
                className={style.icon}
                color={darkColor}
                icon="performanceIndicator"
                size={24}
              />

              <span className={style.text}>Performance indicators</span>
            </div>
          </div>
        )}

        {props.isAllCitiesPage && !props.isContextPage && (
          <div
            className={style.spacer}
            style={{ background: mediumColor }}
          />
        )}

        {props.isAllCitiesPage && (
          <div className={style.rangeWrapper}>
            <div className={style.rangeText}>
              Indicator range for {props.cities.length} Smart cities
            </div>

            <div className={style.rangeIconsWrapper}>
              <div className={style.rangeIconWrapper}>
                <Icon
                  className={style.icon}
                  color={darkColor}
                  icon="maxBarsInCircle"
                  size={24}
                />

                <span className={style.text}>Highest</span>
              </div>

              <div className={style.rangeIconWrapper}>
                <Icon
                  className={style.icon}
                  color={darkColor}
                  icon="minBarsInCircle"
                  size={24}
                />

                <span className={style.text}>Lowest</span>
              </div>
            </div>
          </div>
        )}
      </div>
    </div>
  );
};

PageLegend.propTypes = {
  cities: PropTypes.array, // eslint-disable-line react/forbid-prop-types
  colorName: PropTypes.string.isRequired,
  isAllCitiesPage: PropTypes.bool,
  isContextPage: PropTypes.bool,
};

export default PageLegend;
