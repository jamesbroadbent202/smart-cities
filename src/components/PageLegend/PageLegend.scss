@import '~style/variables';

.wrapper {
  padding-top: 16px;
  padding-bottom: 16px;

  @include for-tablet-landscape-up {
    padding-top: 32px;
    padding-bottom: 32px;
  }
}

.container {
  @include container;

  display: flex;
  align-items: center;
  flex-direction: column;
  font-size: 12px;
  letter-spacing: 0.6px;

  @include for-tablet-portrait-up {
    align-items: flex-end;
  }

  @include for-desktop-up {
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
  }
}

.icon {
  margin-right: 5px;
}

.text {
  vertical-align: middle;
}

.indicatorTypesWrapper {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: flex-end;

  @include for-tablet-portrait-up {
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
  }
}

.indicatorTypeWrapper {
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    &:first-child {
      margin-right: 24px;
    }
  }

  @include for-desktop-up {
    margin-right: 24px;
    margin-bottom: 0;
  }
}

.spacer {
  display: none;

  @include for-desktop-up {
    display: block;
    width: 2px;
    height: 24px;
    margin-right: 24px;
  }
}

.rangeWrapper {
  display: flex;
  flex-direction: column;
  align-items: column;

  @include for-tablet-portrait-up {
    flex-direction: row;
    align-items: center;
    justify-content: flex-end;
  }
}

.rangeIconsWrapper {
  display: flex;
  justify-content: center;
}

.rangeText {
  margin-bottom: 16px;

  @include for-tablet-portrait-up {
    margin-bottom: 0;
    margin-right: 24px;
  }
}

.rangeIconWrapper {
  &:first-of-type {
    margin-right: 24px;
  }
}
