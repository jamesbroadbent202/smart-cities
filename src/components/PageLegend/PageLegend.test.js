import React from 'react';
import { shallow } from 'enzyme';
import PageLegend from './PageLegend';

jest.mock('../Icon/Icon', () => 'Icon');
jest.mock('../Tooltip/Tooltip', () => 'Tooltip');

const defaultProps = {
  colorName: 'jobs',
};

it('should match Snapshot', () => {
  const component = shallow(
    <PageLegend {...defaultProps} />
  );

  expect(component.debug()).toMatchSnapshot();
});

it('should show the "performance indicators" text when not on the context page', () => {
  const component = shallow(
    <PageLegend
      {...defaultProps}
      isContextPage={false}
    />
  );

  expect(component.find('.indicatorTypesWrapper').length).toBe(1);
});

it('should not show the "performance indicators" text when on the context page', () => {
  const component = shallow(
    <PageLegend
      {...defaultProps}
      isContextPage
    />
  );

  expect(component.find('.indicatorTypesWrapper').length).toBe(0);
});

it('should show the range icons on the all cities pages', () => {
  const component = shallow(
    <PageLegend
      {...defaultProps}
      isAllCitiesPage={true}
      cities={[]}
    />
  );

  expect(component.find('.rangeWrapper').length).toBe(1);
});

it('should not show the range icons on a city page', () => {
  const component = shallow(
    <PageLegend
      {...defaultProps}
      isAllCitiesPage={false}
    />
  );

  expect(component.find('.rangeWrapper').length).toBe(0);
});
