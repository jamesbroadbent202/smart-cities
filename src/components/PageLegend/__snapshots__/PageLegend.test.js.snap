// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match Snapshot 1`] = `
"<div className=\\"wrapper\\" style={{...}}>
  <div className=\\"container\\">
    <div className=\\"indicatorTypesWrapper\\">
      <div className=\\"indicatorTypeWrapper\\">
        <Tooltip text=\\"Contextual indicators highlight the characteristics of a city that are not amenable to or potentially appropriate for local policy intervention but they can help to explain why a city performs the way it does.\\" borderColor=\\"JOBS_900\\">
          <Icon className=\\"icon\\" color=\\"JOBS_900\\" icon=\\"contextualIndicator\\" size={24} />
          <span className=\\"text\\">
            Contextual indicators
          </span>
        </Tooltip>
      </div>
      <div className=\\"indicatorTypeWrapper\\">
        <Tooltip text=\\"A performance indicator is a measurable value that demonstrates how effectively cities are performing against the Australian Governmentâ€™s Smart Cities policy priorities.\\" borderColor=\\"JOBS_900\\">
          <Icon className=\\"icon\\" color=\\"JOBS_900\\" icon=\\"performanceIndicator\\" size={24} />
          <span className=\\"text\\">
            Performance indicators
          </span>
        </Tooltip>
      </div>
    </div>
  </div>
</div>"
`;
